src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs

```diff
@@ -1109,6 +1109,14 @@ private void WorkspaceAdded(WorkspaceModel item)
             else
             {
                 var newVm = new WorkspaceViewModel(item, this);
+
+                // For Json Workspaces, workspace view info need to be read agin from file
+                string fileContents;
+                if (DynamoUtilities.PathHelper.isValidJson(newVm.Model.FileName, out fileContents))
```

Comment:
also, how slow is this check json function?

Reply:
In this case, I have to check because I have to gurantee that file is still valid at the moment user double click on CustomNode to open CustomNodeWorkspace.

Overall, from testing I have not seen anything obvious yet